tools/flight_recorder/fr_trace.py

```diff
@@ -170,6 +187,50 @@ class Database(NamedTuple):
 }
 
 
+class MatchState(Enum):
+    """
+    Enum representing the possible states of matching for collective operations.
+
+    - FULLY_MATCHED: Indicates that all aspects of the collective operations match.
+    - COLLECTIVE_TYPE_MISMATCH: The types of the collective operations differ.
+    - SIZE_OR_SYNTAX_MISMATCH: There is a mismatch in input/output sizes or violation of collective syntax.
+    - COLLECTIVE_STATE_MISMATCH:
+        The states of the collective not same, such as one finished while another just started or scheduled.
+    - UNDECIDED:
+        The match status is ambiguous or cannot be determined, e.g., we might need to check all ranks for alltoall_base.
+    """
+
+    FULLY_MATCHED = 1
+    COLLECTIVE_TYPE_MISMATCH = 2
+    SIZE_OR_SYNTAX_MISMATCH = 3
+    COLLECTIVE_STATE_MISMATCH = 4
+    UNDECIDED = 5
+
+
+def check_size_even_expand(list1: List[Any], list2: List[Any], size: int) -> bool:
```

Comment:
i think you're referring to 'broadcasting' by "expand" - maybe rename? (and maybe there is a torch helper existing for checking size including broadcasting? but no big deal..

Reply:
Oh I didn't see any existing helper function, yes, let me rename it.